fedimint/src/bin/fedimintd.rs

```diff
@@ -9,17 +9,51 @@ use tracing_subscriber::EnvFilter;
 use tracing_subscriber::Layer;
 
 #[derive(Parser)]
-pub struct ServerOpts {
-    pub cfg_path: PathBuf,
-    pub db_path: PathBuf,
-    #[cfg(feature = "telemetry")]
-    #[clap(long)]
-    pub with_telemetry: bool,
+struct Cli {
+    #[clap(subcommand)]
+    command: Command,
+}
+#[derive(Subcommand)]
+enum Command {
+    VersionHash,
+    Start {
+        cfg_path: PathBuf,
+        db_path: PathBuf,
+        #[clap(long, takes_value = false)]
+        with_telemetry: bool,
+    },
```

Comment:
Could we instead add an optional `--version` command which would print out the version information?

We shouldn't add `start` commands to everything because that would be very unexpected. You don't type `bitcoind start` to run bitcoind. You just type `bitcoind` ...

Reply:
I will not use the parser struct for the optional flag though becasue it would be tedious for users to supply all other required args just to run `version-hash`